### ISSUE TITLE
is copyright notice generation correct?

### ISSUE THREAD

**Issue by sandrotosi** [PyCQA/pylint#1808]
while packaging 1.8.1 for debian i noticed this:

```
diff -urNad pylint-1.7.4/pylint/checkers/design_analysis.py pylint-1.8.1/pylint/checkers/design_analysis.py
--- pylint-1.7.4/pylint/checkers/design_analysis.py     2017-09-30 04:10:24.000000000 -0400
+++ pylint-1.8.1/pylint/checkers/design_analysis.py     2017-12-15 06:21:24.000000000 -0500
@@ -1,5 +1,11 @@
+# -*- coding: utf-8 -*-
 # Copyright (c) 2006, 2009-2010, 2012-2015 LOGILAB S.A. (Paris, FRANCE) <contact@logilab.fr>
-# Copyright (c) 2014-2016 Claudiu Popa <pcmanticore@gmail.com>
+# Copyright (c) 2012, 2014 Google, Inc.
+# Copyright (c) 2014-2017 Claudiu Popa <pcmanticore@gmail.com>
+# Copyright (c) 2014 Arun Persaud <arun@nubati.net>
+# Copyright (c) 2015 Ionel Cristian Maries <contact@ionelmc.ro>
+# Copyright (c) 2016 Łukasz Rogalski <rogalski.91@gmail.com>
+# Copyright (c) 2017 ahirnish <ahirnish@gmail.com>
 
 # Licensed under the GPL: https://www.gnu.org/licenses/old-licenses/gpl-2.0.html
 # For details: https://github.com/PyCQA/pylint/blob/master/COPYING
@@ -211,7 +217,7 @@
 
     @check_messages('too-many-return-statements', 'too-many-branches',
                     'too-many-arguments', 'too-many-locals',
-                    'too-many-statements')
+                    'too-many-statements', 'keyword-arg-before-vararg')
     def visit_functiondef(self, node):
         """check function name, docstring, arguments, redefinition,
         variable names, max locals
```

tbh i find it really hard to believe the addition of a single value in a decorator can produce the addition of 5 people to the copyright notice for that specific file.

Debian requires to keep track of all the copyright holder (and license) for all the files in the tools we package, and keeping track of all the changes in pylint is extermely time-consuming (like 1+ hrs every time a new pylint is released); so i'm wondering if the process you use to generate the copyright notice is accurate or it may require some adjustments.

thanks for considering,
Sandro

**Comment by PCManticore** [comment]
This is definitely possible and I hope that we'll get to a point where this won't be a problem for Debian packaging any longer. As mentioned in previous discussions, we are using a tool to retrieve the authors of a file (https://github.com/PCManticore/copyrite). It uses two values for determining the authorship, the number of changes and the minimum number of lines touched. To be honest, I'm not entirely sure that this is 100% correct, but I wanted to avoid having a changed comment resulting in a new copyright notice. Recently I pushed to the repository the threshold values I used for the last release. These are different than those from the previous releases, since I changed machines in the mean time and I forgot what threshold values I was using: https://github.com/PyCQA/pylint/blob/master/doc/release.txt#L63. Since this is now in our release process, it means that they could be reproducible for the next releases. Given this, let me know if there is something that we might improve here to help you with packaging.
